src/Agent.Listener/Configuration/ConfigurationProvider.cs

```diff
@@ -234,25 +233,22 @@ public string GetServerUrl(CommandSettings command)
 
         public async Task<int> GetPoolId(CommandSettings command)
         {
-            int poolId = 0;
+            int poolId;
             while (true)
             {
                 _machineGroupName = command.GetMachineGroupName();
                 try
                 {
                     poolId =  await GetPoolIdAsync(_projectName, _machineGroupName);
+                    Trace.Info($"PoolId for machine group '{_machineGroupName}' is '{poolId}'.");
+                    break;
                 }
                 catch (Exception e) when (!command.Unattended)
                 {
                     _term.WriteError(e);
                 }
 
-                if (poolId > 0)
-                {
-                    break;
-                }
-
-                _term.WriteError(StringUtil.Loc("FailedToFindPool"));
+                _term.WriteError(StringUtil.Loc("FailedToFindMachineGroup"));
```

Comment:
If failed to find machine group - that error would have been come as exception and printed in the catch block above. This is really pool not found. I will suggest, for now throw Pool error only. We will discuss and can change it later